include/nbl/builtin/glsl/format/encode.glsl

```diff
@@ -82,4 +82,51 @@ uint nbl_glsl_encodeRGB10A2_SNORM(in vec4 col)
 	return quantized.r|quantized.g|quantized.b|quantized.a;
 }
 
+// TODO: break it down into uint nbl_glsl_encode_ufloat_exponent(in float _f32) and nbl_glsl_encode_ufloat_mantissa(in float _f32, in uint mantissaBits, in bool leadingOne)
+uint nbl_glsl_encode_ufloat(in float _f32, in uint mantissaBits)
+{
+	uint minSinglePrecisionVal = floatBitsToUint(nbl_glsl_R11G11B10_MIN);
+	uint maxSinglePrecisionVal = floatBitsToUint(nbl_glsl_R11G11B10_MAX);
+
+	if (_f32 < uintBitsToFloat(maxSinglePrecisionVal))
+	{
+		if (_f32 < uintBitsToFloat(minSinglePrecisionVal))
+			return 0;
+
+		const int exp = int(nbl_glsl_ieee754_extract_biased_exponent(_f32) - 127);
```

Comment:
seems you also need a `nbl_glsl_ieee754_extract_exponent` so we can get rid of that nasty handrolled `-127`

Reply:
Resolved.